### PR TITLE
improve ref doc, split samples to guide

### PR DESCRIPTION
**Pull request by kkraune**
The guide is ugly and needs more work, but want to get the improved reference in. There is probably things still missing there, we can review later

### CODE REVIEW

File: documentation/reference/document-v1-api-reference.html [hunk 1]
```diff
@@ -1,212 +1,342 @@
...
+        </td>
+      </tr>
+      <tr>
+        <th>Visit</th>
+        <td>
+          <p>
+            Visit iterates over all document or a set of documents, in chunks,
+            using <a href="#continuation">continuation</a> as a progress token.
+          </p><p>
+          A Document API request can only retrieve data from one content cluster,
```
Comment (jonmv) [inline documentation/reference/document-v1-api-reference.html:62]:
```suggestion
          A Document API visit request can only retrieve data from one content cluster,
```

File: documentation/reference/document-v1-api-reference.html [hunk 2]
```diff
@@ -1,212 +1,342 @@
...
+/document/v1/&lt;namespace&gt;/&lt;document-type&gt;/number/&lt;text-group-id&gt;
 </pre>
-  Visit documents across all <em>non-global</em> document types and namespaces stored
-  in content cluster <code>mycluster</code>:
+            </li>
+          </ul>
+          <p>
+            Parent documents are <a href="services-content.html#document">global</a>
+            and in the <code>global</code> <a href="#bucketspace">bucket space</a>.
+            By default, visit will visit non-global documents
```
Comment (jonmv) [inline documentation/reference/document-v1-api-reference.html:106]:
```suggestion
            By default, visit at the root level will visit non-global documents
```

File: documentation/reference/document-v1-api-reference.html [hunk 3]
```diff
@@ -1,212 +1,342 @@
...
+  <th>fieldSet</th>
+  <td>String</td>
+  <td>A <a href="../documents.html#fieldsets">field set string</a>
+    with the set of document fields to return.
+    Default is <code>&lt;visited document type&gt;:[document]</code>, returning all fields.
+  </td>
+</tr><tr id="route">
+  <th>route</th>
+  <td>String</td>
+  <td>The route for document operations. Default value is <code>default</code>.
```
Comment (jonmv) [inline documentation/reference/document-v1-api-reference.html:206]: Default is actually `default-get` for DocumentGet operations, and something very custom for visitors.

File: documentation/reference/document-v1-api-reference.html [hunk 4]
```diff
@@ -1,212 +1,342 @@
...
+    See <a href="../routing.html">routes</a>.</td>
+</tr><tr id="selection">
+  <th>selection</th>
+  <td>String</td>
+  <td>Select a set of documents using <a href="../content/visiting.html">visiting</a> -
```
Comment (jonmv) [inline documentation/reference/document-v1-api-reference.html:211]:
```suggestion
  <td>Select only a subset of documents when <a href="../content/visiting.html">visiting</a> -
```

File: documentation/reference/document-v1-api-reference.html [hunk 5]
```diff
@@ -1,212 +1,342 @@
...
+  </td>
+</tr>
 </tbody>
 </table>
+
+
+
+<h2 id="request-body">Request body</h2>
+<p>
+  POST and PUT has a body.
```
Comment (jonmv) [inline documentation/reference/document-v1-api-reference.html:250]:
```suggestion
  POST and PUT have a body.
```

File: documentation/reference/document-v1-api-reference.html [hunk 6]
```diff
@@ -1,212 +1,342 @@
...
+  <td>Gateway timeout.</td>
+</tr>
+</table>
+
+
+
+<h2 id="response-format">Response format</h2>
+<p>
+  Documents and operations use the <a href="document-json-format.html">document format</a>.
+  The document fields must match the <a href="../schemas.html">schema</a>.
```
Comment (jonmv) [inline documentation/reference/document-v1-api-reference.html:298]: This should rather be specified where you write about POST and PUT bodies — _we_ ensure they match the schema here :)

File: documentation/reference/document-v1-api-reference.html [hunk 7]
```diff
@@ -1,212 +1,342 @@
...
+</p><p>
+  Responses always include the <em>path</em> used in the request and one of:
```
Comment (jonmv) [inline documentation/reference/document-v1-api-reference.html:300]:
```suggestion
  Successful responses always include the <em>path</em> used in the request and one of:
```

File: documentation/reference/document-v1-api-reference.html [hunk 8]
```diff
@@ -1,212 +1,342 @@
...
+    ],
+    "continuation": "&lt;continuation string&gt;"
+}
+</pre>
+</p><p>
+  A <code>message</code> can be returned for failed operations:
+<pre>
+{
+    "pathId":  "&lt;pathid&gt;",
+    "id":      "&lt;id&gt;",
```
Comment (jonmv) [inline documentation/reference/document-v1-api-reference.html:338]: May or may not be there, so 
```suggestion
```

**Review by jonmv — APPROVED** [review]
Nice :) Some comments.

File: documentation/reference/document-v1-api-reference.html [hunk 9]
```diff
@@ -1,212 +1,342 @@
...
+&lt;services&gt;
+  &lt;container&gt;
+    <span style="background-color: yellow;">&lt;document-api/&gt;</span>
 </pre>
 </p>
 
 
 
-<h2 id="operations">Operations</h2>
+<h2 id="request">Request</h2>
```
[the remark this answers is not in the document]
Comment (kkraune) [reply documentation/reference/document-v1-api-reference.html:29]: ok

File: documentation/reference/document-v1-api-reference.html (hunk 4 above)
Comment (kkraune) [reply documentation/reference/document-v1-api-reference.html:211]: ok

File: documentation/reference/document-v1-api-reference.html (hunk 5 above)
Comment (kkraune) [reply documentation/reference/document-v1-api-reference.html:250]: yup

File: documentation/reference/document-v1-api-reference.html (hunk 6 above)
Comment (kkraune) [reply documentation/reference/document-v1-api-reference.html:298]: correct! moving

File: documentation/document-v1-api-guide.html [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+
+
+
+<h2 id="using-fieldsets">Using fieldsets</h2>
+<p>
+  <!-- ToDo: rewrite this -->
+  When visiting across all document types,
+  some internal document fields (e.g. Geo fields) set by Vespa may be returned as part of the results.
+  To avoid this, limit visiting to just one document type using <em>selection</em>
+  and explicitly filter these internal fields away using <em>fieldSet</em>:
```
[the remark this answers is not in the document]
Comment (kkraune) [reply documentation/document-v1-api-guide.html:129]: yes I left a todo to make all of this more intuitive, I will do some more testing with multiple doc types and fix in later PR

**Pull request merged**